src/Engine/ProtoCore/DSASM/Executive.cs

```diff
@@ -6928,30 +6928,54 @@ private void DEP_Handler(Instruction instruction)
             {
                 // Go to the next pc
                 pc++;
+                Properties.executingGraphNode = GetNextGraphNodeToExecute(pc, ci, fi);
+                if (Properties.executingGraphNode != null)
+                {
+                    Properties.executingGraphNode.isDirty = false;
+                    pc = Properties.executingGraphNode.updateBlock.startpc;
+                }
+            }
+            GC();
+            return;
+        }
 
-                // Given the next pc, get the next graphnode to execute and mark it clean
-                if (runtimeCore.Options.IsDeltaExecution)
+        /// <summary>
+        /// Get the next graphnode to execute given the current next pc and scope
+        /// </summary>
+        /// <param name="pc"></param>
+        /// <param name="ci"></param>
+        /// <param name="fi"></param>
+        private AssociativeGraph.GraphNode GetNextGraphNodeToExecute(int nextPC, int ci, int fi)
+        {
+            AssociativeGraph.GraphNode nextGraphNode = null;
+
+            // Given the next pc, get the next graphnode to execute and mark it clean
+            if (runtimeCore.Options.IsDeltaExecution)
+            {
+                if (IsGlobalScope())
```

Comment:
Is there any chance that ``IsGlobalScope()`` return ture **and** ``runtimeCore.Options.ApplyUpdate`` is false? My understanding is ``ApplyUpdate`` will be set to true **in global scope**.